pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -768,6 +814,85 @@ impl RemoteTimelineClient {
+            let mut locked = self.upload_queue.lock().unwrap();
+            let stopped = match &mut *locked {
+                UploadQueue::Uninitialized => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Initialized(_) => {
+                    anyhow::bail!("upload queue is not stopped but initialized")
+                }
+                UploadQueue::Stopped(stopped) => stopped,
+            };
+
+            if !matches!(stopped.deleted_at, SetDeletedFlagProgress::Successful(_)) {
+                anyhow::bail!("deleted_at is not set")
+            }
+
+            debug_assert!(stopped
+                .upload_queue_for_deletion
+                .inprogress_tasks
+                .is_empty());
+            debug_assert!(stopped
+                .upload_queue_for_deletion
+                .queued_operations
+                .is_empty());
+
+            // schedule the actual deletions
+            for name in stopped.upload_queue_for_deletion.latest_files.keys() {
+                let op = UploadOp::Delete(RemoteOpFileKind::Layer, name.clone());
+                self.calls_unfinished_metric_begin(&op);
+                stopped
+                    .upload_queue_for_deletion
+                    .queued_operations
+                    .push_back(op);
+                info!("scheduled layer file deletion {}", name.file_name());
+            }
+
+            self.launch_queued_tasks(&mut stopped.upload_queue_for_deletion);
+
+            self.schedule_barrier(&mut stopped.upload_queue_for_deletion)
+        };
+
+        receiver.changed().await?;
+
+        // Do not delete index part yet, it is needed for possible retry. If we remove it first
+        // and retry will arrive to different pageserver there wont be any traces of it on remote storage
+        let timeline_path = self.conf.timeline_path(&self.timeline_id, &self.tenant_id);
+        let timeline_storage_path = self.conf.remote_path(&timeline_path)?;
+
+        let remaining = self
+            .storage_impl
+            .list_prefixes(Some(&timeline_storage_path))
+            .await?;
+
+        let remaining: Vec<RemotePath> = remaining
+            .into_iter()
+            .filter(|p| p.object_name() != Some(IndexPart::FILE_NAME))
+            .collect();
+
+        if !remaining.is_empty() {
+            warn!(
+                "Found {} files not bound to index_file.json, proceeding with their deletion",
+                remaining.len()
+            );
+            for file in remaining {
+                warn!("Removing {}", file.object_name().unwrap_or_default());
+                self.storage_impl.delete(&file).await?;
+            }
+        }
+
+        let index_file_path = timeline_storage_path.join(Path::new(IndexPart::FILE_NAME));
+        info!("deleting index part");
+        self.storage_impl.delete(&index_file_path).await?;
+
+        info!("Done");
```

Comment:
```suggestion
        debug!("Done");
```

I always have to filter all these out from logs (same with starting) so I'd prefer if we didn't add more. I do understand how does it help with initial debugging but still, we have so many of these.

Reply:
Maybe the index part even, or have a combined single `info!` about how many we deleted:

```
info!(layer_files, other_files, "done deleting, including index_part.json");
```

Reply:
Done